src/repository/opamDownload.ml

```diff
@@ -283,13 +286,24 @@ module SWHID = struct
 
   let instance = OpamUrl.of_string "https://archive.softwareheritage.org"
   (* we keep api 1 hardcoded for the moment *)
-  let full_url middle hash = OpamUrl.Op.(instance / "api" / "1" / middle / hash / "")
+  let full_url middle hash =
+    OpamUrl.Op.(instance / "api" / "1" / middle / hash / "")
+  let vault_url kind hash =
+    full_url ("vault/" ^ kind) ("swh:1:dir:" ^ hash)
 
-  let check_liveness () =
-    OpamProcess.Job.catch (fun _ -> Done false)
-    @@ fun () ->
-      get_output ~post:true OpamUrl.Op.(instance / "api" / "1" / "ping" / "")
-      @@| fun _ -> true
+  let fallback_err fmt = Printf.sprintf ("SWH fallback: "^^fmt)
+
+  let get_output ?(post=false) url =
+    get_output ~post url @@| function
+    | Some out -> out
+    | None ->
+      (* Shouldn't happen, we already checked that a post tool is used *)
+      (* XXX change to an assert false ? *)
```

Comment:
I don't think this display is needed, as in this module the main entry `archive_fallback` checks that.

Reply:
in that case i agree this could be replaced by `assert false`